packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+import {
+	type ImplicitFieldSchema,
+	type ITree,
+	SharedTree,
+	type TreeView,
+	type TreeViewConfiguration,
+} from "@fluidframework/tree/internal";
+
+import { PureDataObject } from "./pureDataObject.js";
+
+/**
+ * Channel ID of {@link TreeDataObject}'s root {@link @fluidframework/tree#SharedTree}.
+ */
+const treeChannelId = "tree-data-object";
+
+const uninitializedErrorString =
+	"The tree has not yet been initialized. The data object must be initialized before accessing.";
+
+/**
+ * {@link @fluidframework/tree#SharedTree}-backed {@link PureDataObject | data object}.
+ *
+ * @remarks
+ *
+ * Note: to initialize the tree's data for initial creation, implementers of this class will need to override {@link PureDataObject.initializingFirstTime} and set the data in {@link TreeDataObject.tree}.
+ *
+ * @example Implementing `initializingFirstTime`
+ *
+ * ```typescript
+ * protected override async initializingFirstTime(): Promise<void> {
+ * 	this.tree.initialize(...);
+ * }
+ * ```
+ *
+ * @privateRemarks
+ * TODO: Before promoting this beyond internal, we should consider alternative API patterns that don't depend on
+ * sub-classing and don't leak Fluid concepts that should ideally be internal.
+ * See `tree-react-api` for an example of a pattern that avoids unnecessary leakage of implementation details.
+ *
+ * @internal
+ */
+export abstract class TreeDataObject<
+	TSchema extends ImplicitFieldSchema = ImplicitFieldSchema,
+> extends PureDataObject {
+	/**
+	 * The configuration used to initialize new documents, as well as to interpret (schematize) existing ones.
+	 *
+	 * @remarks
+	 * The fact that a single view schema is provided here (on the data object) makes it impossible to try and apply multiple different schema.
+	 * Since the view schema currently does not provide any adapters for handling differences between view and stored schema,
+	 * it's also impossible for this single view schema to handle multiple different stored schema.
+	 * Therefore, with this current API, two different applications (or different versions of the same application)
+	 * with differing stored schema requirements (as implied by their view schema) can not collaborate on the same tree.
+	 * The only schema evolution that's currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
+	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
+	 * Future work on tree as well as these utilities should address this limitation.
+	 */
+	public abstract readonly config: TreeViewConfiguration<TSchema>;
+
+	/**
+	 * View of the underlying tree.
+	 */
+	#treeView: TreeView<TSchema> | undefined;
```

Comment:
It should be easy to fix the issues in that big disclaimer (put them pack on the react tree view), and gain a lot more flexibility.
```suggestion
> extends PureDataObject<TView> {
	/**
	 * Generates the "view" exposed via the {@link PureDataObject.tree} property.
	 */
	protected abstract generateView(tree: ITree): TView;

	/**
	 * View of the underlying tree.
	 */
	#treeView: TView | undefined;
```

Maybe also rename `PureDataObject.tree` to `PureDataObject.view`.

Reply:
Edit: I misread the code block. I see that this would be called once internally. Fair enough. I can play around with this.